Index: debathena-get-message-9.4.0/view_message_by_zephyr.c
===================================================================
--- debathena-get-message-9.4.0.orig/view_message_by_zephyr.c	2007-02-04 01:48:18.000000000 -0500
+++ debathena-get-message-9.4.0/view_message_by_zephyr.c	2007-02-04 01:48:24.000000000 -0500
@@ -50,7 +50,7 @@
   /* check that there is *something* after the headers */
   if(*ptr) {
     /* don't even fork... this just exits anyway... */
-    execl("/usr/athena/bin/zwrite",
+    execl("/usr/bin/zwrite",
 	  "zwrite", "-d", "-q", "-n",  whoami, "-m", ptr, 0);
     /* put logging here in case the exec fails. */
     syslog(LOG_INFO, "GMS client execl of zwrite failed [%s]",
