Index: debathena-finger-9.4.10/finger.c
===================================================================
--- debathena-finger-9.4.10.orig/finger.c	2007-08-10 02:08:05.000000000 -0400
+++ debathena-finger-9.4.10/finger.c	2007-08-10 02:08:27.000000000 -0400
@@ -181,7 +181,7 @@
 int hack = 1;			/* -h option default */
 int idlep = 0;			/* -i option default */
 int large = 0;			/* -l option default */
-int match = 1;			/* -m option default */
+int match = 0;			/* -m option default */
 int plan = 1;			/* -p option default */
 int unquick = 1;		/* -q option default */
 int small = 0;			/* -s option default */
