Index: debathena-libgms-9.4.24/globalmessage.h
===================================================================
--- debathena-libgms-9.4.24.orig/globalmessage.h	2007-08-16 00:45:40.000000000 -0400
+++ debathena-libgms-9.4.24/globalmessage.h	2007-08-16 00:55:05.000000000 -0400
@@ -33,7 +33,7 @@
 #define GMS_SERV_NAME "globalmessage"
 #define GMS_SERV_PROTO "udp"
 
-#define GMS_MESSAGE_NAME "/usr/tmp/.messages"
+#define GMS_MESSAGE_NAME "/var/cache/debathena-libgms/messages"
 
 #define GMS_USERFILE_NAME "/.message_times"
 #define GMS_USERFILE_NAME_LEN (sizeof(GMS_USERFILE_NAME)-1)
