Index: debathena-dotfiles-9.4.43/bash_login
===================================================================
--- debathena-dotfiles-9.4.43.orig/bash_login	2008-01-22 21:59:06.000000000 -0500
+++ debathena-dotfiles-9.4.43/bash_login	2008-01-22 21:59:06.000000000 -0500
@@ -19,14 +19,6 @@
 ATHENA_LOGIN_SESSION=$$
 export ATHENA_LOGIN_SESSION
 
-# Create a unique temporary directory for the session, and remember
-# the directory name in the environment.
-sessiondir="`$initdir/mksessiondir -s $$`"
-if [ -n "$sessiondir" ]; then
-  ATHENA_SESSION_TMPDIR="$sessiondir"
-  export ATHENA_SESSION_TMPDIR
-fi
-
 if [ ! -r ~/.hushlogin ]; then
 	verbose_login=t		# Echo messages unless user has ~/.hushlogin.
 fi
Index: debathena-dotfiles-9.4.43/login
===================================================================
--- debathena-dotfiles-9.4.43.orig/login	2008-01-22 21:59:06.000000000 -0500
+++ debathena-dotfiles-9.4.43/login	2008-01-22 21:59:06.000000000 -0500
@@ -13,13 +13,6 @@
 
 set initdir=/usr/athena/lib/init
 
-# Create a unique temporary directory for the session, and remember
-# the directory name in the environment.
-set sessiondir = "`$initdir/mksessiondir -s $$`"
-if ("$sessiondir" != "") then
-  setenv ATHENA_SESSION_TMPDIR "$sessiondir"
-endif
-
 if (! -r ~/.hushlogin) then
 	set verbose_login	# Echo messages unless user has ~/.hushlogin.
 endif
Index: debathena-dotfiles-9.4.43/Xsession
===================================================================
--- debathena-dotfiles-9.4.43.orig/Xsession	2008-01-22 21:59:18.000000000 -0500
+++ debathena-dotfiles-9.4.43/Xsession	2008-01-22 21:59:25.000000000 -0500
@@ -54,14 +54,6 @@
 # Set X screensaver to noblanking to avoid power saving.
 xset s noblank
 
-# Create a unique temporary directory for the session, and remember
-# the directory name in the environment.
-sessiondir="`$initlib/mksessiondir -s $$`"
-if [ -n "$sessiondir" ]; then
-	ATHENA_SESSION_TMPDIR="$sessiondir"
-	export ATHENA_SESSION_TMPDIR
-fi
-
 # Only attempt using bugme if packs are actually available.
 if [ -x /usr/athena/bin/bugme ]; then
 	/usr/athena/bin/bugme $session
