Index: debathena-dotfiles-9.4.34+cvs20070823/Makefile
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/Makefile	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/Makefile	2007-11-15 00:50:01.000000000 -0500
@@ -2,9 +2,9 @@
 
 SHELL=/bin/sh
 PROTOTYPE=/usr/prototype_user
-PROTOTMP=/usr/athena/lib/prototype_tmpuser
-SYSTEM=/usr/athena/lib/init
-ATHMANDIR=/usr/athena/man
+PROTOTMP=/usr/lib/prototype_tmpuser
+SYSTEM=/usr/lib/init
+ATHMANDIR=/usr/share/man
 ATHRETCDIR=/etc/athena
 ATHLOGINDIR=${ATHRETCDIR}/login
 
Index: debathena-dotfiles-9.4.34+cvs20070823/gnome/Makefile
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/gnome/Makefile	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/gnome/Makefile	2007-11-15 00:50:01.000000000 -0500
@@ -1,10 +1,10 @@
 # $Id: Makefile,v 1.6 2006-08-09 15:42:50 rbasch Exp $
 
-PIXMAPDIR=/usr/athena/share/pixmaps
-GNOMEATHDIR=/usr/athena/share/gnome/athena
-SCHEMASDIR=/etc/athena/gconf/schemas
+PIXMAPDIR=/usr/share/pixmaps
+GNOMEATHDIR=/usr/share/gnome/athena
+SCHEMASDIR=/usr/share/gconf/schemas
 AFSMENUS=/afs/athena.mit.edu/system/config/panel/menus
-APPLICATIONS=/usr/athena/share/applications
+APPLICATIONS=/usr/share/applications
 
 all: athena.schemas
 
Index: debathena-dotfiles-9.4.34+cvs20070823/bashrc
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/bashrc	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/bashrc	2007-11-15 00:50:11.000000000 -0500
@@ -55,11 +55,14 @@
 		MANPATH=$MANPATH:/usr/sfw/man
 		;;
 	linux)
-		athena_path=/usr/athena/bin:/usr/athena/etc
-		athena_path=$athena_path:/bin/athena:/usr/bin:/bin
-		athena_path=$athena_path:/usr/X11R6/bin:/usr/athena/etc
-		athena_path=$athena_path:/usr/sbin:/sbin
-		MANPATH=/usr/athena/man:/usr/share/man:/usr/X11R6/man
+		athena_path=/usr/local/bin:/usr/bin:/bin:/usr/games
+		MANPATH=/usr/share/man
+		if [ -d /usr/bin/X11 ]; then
+		    athena_path=$athena_path:/usr/bin/X11
+		fi
+		if [ -d /usr/X11R6/man ]; then
+		    MANPATH=$MANPATH:/usr/X11R6/man
+		fi
 		;;
 	*)
 		echo "Standard dotfiles do not support system type $HOSTTYPE."
@@ -91,12 +94,6 @@
 		export OPENWINHOME=/usr/openwin
 		;;
 	linux)
-		# This is to ensure that native programs use the Athena
-		# gconf, ORBit, bonobo-activation, etc. libraries
-		# rather than the native ones.  GNOME programs outside
-		# of /usr/athena/bin may not function properly if it is
-		# not set.
-		export LD_LIBRARY_PATH=/usr/athena/lib
 		;;
 	esac
 	
@@ -187,10 +184,10 @@
 	fi
 	case $HOME in
 	/var/athena/tmphomedir/*)
-		PS1="athena (temporary homedir)\$ "
+		PS1="debathena (temporary homedir)\$ "
 		;;
 	*)
-		PS1="athena\$ "
+		PS1="debathena\$ "
 		;;
 	esac
 	CDPATH=.:~
Index: debathena-dotfiles-9.4.34+cvs20070823/cshrc
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/cshrc	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/cshrc	2007-11-15 00:50:11.000000000 -0500
@@ -37,9 +37,10 @@
   set athena_manpath=${athena_manpath}:/usr/sfw/man
   breaksw
 case linux:
-  set athena_path=( /usr/athena/bin /usr/athena/etc /bin/athena /usr/bin \
-		    /bin /usr/X11R6/bin /usr/sbin /sbin )
-  set athena_manpath=/usr/athena/man:/usr/share/man:/usr/X11R6/man
+  set athena_path=( /usr/local/bin /usr/bin /bin /usr/games )
+  set athena_manpath=/usr/share/man
+  if ( -d /usr/bin/X11 ) set athena_path=( $athena_path /usr/bin/X11 )
+  if ( -d /usr/X11R6/man ) set athena_manpath=( $athena_manpath /usr/X11R6/man )
   breaksw
 default:
   echo "Standard dotfiles do not support system type $HOSTTYPE."
@@ -74,11 +75,6 @@
     setenv OPENWINHOME /usr/openwin
     breaksw
   case linux:
-    # This is to ensure that native programs use the Athena gconf, ORBit,
-    # bonobo-activation, etc. libraries rather than the native ones.
-    # GNOME programs outside of /usr/athena/bin may not function properly
-    # if it is not set.
-    setenv LD_LIBRARY_PATH /usr/athena/lib
     breaksw
   endsw
 
@@ -199,9 +195,9 @@
   # Set prompt.
   set promptchars="%#"
   if ($home =~ /var/athena/tmphomedir/*) then
-    set prompt = "athena (temporary homedir)%# "
+    set prompt = "debathena (temporary homedir)%# "
   else
-    set prompt = "athena%# "
+    set prompt = "debathena%# "
   endif
 
   set cdpath = (~)		#   Path to search for directory changes
Index: debathena-dotfiles-9.4.34+cvs20070823/mksessiondir.sh
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/mksessiondir.sh	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/mksessiondir.sh	2007-11-15 00:50:02.000000000 -0500
@@ -27,7 +27,7 @@
 # First try to create the default directory name, based only on the
 # user name and session ID.  If that fails, we will use this name as
 # the prefix in finding a unique directory name.
-dir="/tmp/session-${USER}-${sid}"
+dir="/var/run/athena-sessions/session-${USER}-${sid}"
 mkdir -m 700 "$dir" > /dev/null 2>&1
 if [ $? -ne 0 ]; then
   # Default name failed, try to uniquify.
Index: debathena-dotfiles-9.4.34+cvs20070823/bash_login
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/bash_login	2007-11-15 00:49:37.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/bash_login	2007-11-15 00:50:12.000000000 -0500
@@ -112,7 +112,7 @@
 	$ZEPHYR_CLIENT			# Start Zephyr client
 	get_message -new -login		# Display current motd
 	mailquota -n			# Check quota on post office server
-	from -t -n			# Check for mail
+	from.debathena -t -n		# Check for mail
 fi
 
 
Index: debathena-dotfiles-9.4.34+cvs20070823/login
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/login	2007-11-15 00:49:38.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/login	2007-11-15 00:50:12.000000000 -0500
@@ -96,7 +96,7 @@
 	$ZEPHYR_CLIENT			# Start Zephyr client
 	get_message -new -login		# Display current motd
 	mailquota -n			# Check quota on post office server
-	from -t -n			# Check for mail
+	from.debathena -t -n		# Check for mail
 endif
 
 
Index: debathena-dotfiles-9.4.34+cvs20070823/xsession
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/xsession	2007-11-15 00:49:38.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/xsession	2007-11-15 00:50:02.000000000 -0500
@@ -245,7 +245,7 @@
   if (! $?skip_quotawarn) then
     ($initdir/quotawarn &)	# Display warning dialogs if near quota
   endif
-  from -t -n			# Check for mail
+  from.debathena -t -n		# Check for mail
   if (! $?skip_xss) then
     (xscreensaver -no-splash &)
   endif
Index: debathena-dotfiles-9.4.34+cvs20070823/quotawarn.sh
===================================================================
--- debathena-dotfiles-9.4.34+cvs20070823.orig/quotawarn.sh	2007-11-15 00:49:38.000000000 -0500
+++ debathena-dotfiles-9.4.34+cvs20070823/quotawarn.sh	2007-11-15 00:50:02.000000000 -0500
@@ -3,7 +3,7 @@
 # $Id: quotawarn.sh,v 1.4 2007-04-09 17:13:19 ghudson Exp $
 
 # Determine the user's home directory usage and quota.
-qline=`quota -v -f "$USER" | awk '/^\// {print}'`
+qline=`quota.debathena -v -f "$USER" | awk '/^\// {print}'`
 usage=`echo "$qline" | awk '{print $2}'`
 quota=`echo "$qline" | awk '{print $3}'`
 quota90=`expr "${quota:-0}" \* 9 / 10`
