Index: debathena-pyhesiodfs-0.0.r77/pyHesiodFS.py
===================================================================
--- debathena-pyhesiodfs-0.0.r77.orig/pyHesiodFS.py	2008-07-10 21:13:22.000000000 -0400
+++ debathena-pyhesiodfs-0.0.r77/pyHesiodFS.py	2008-07-10 21:13:44.000000000 -0400
@@ -46,10 +46,7 @@
     def __init__(self, *args, **kwargs):
         Fuse.__init__(self, *args, **kwargs)
         self.fuse_args.add("allow_other", True)
-        self.fuse_args.add("noappledouble", True)
-        self.fuse_args.add("noapplexattr", True)
         self.fuse_args.add("fsname", "pyHesiodFS")
-        self.fuse_args.add("volname", "MIT")
         self.mounts = {}
     
     def getattr(self, path):
