Index: debathena-mitmailcreate-9.4.0/mitmailcreate.pl
===================================================================
--- debathena-mitmailcreate-9.4.0.orig/mitmailcreate.pl	2007-01-30 02:48:43.000000000 -0500
+++ debathena-mitmailcreate-9.4.0/mitmailcreate.pl	2007-01-30 02:48:43.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/athena/bin/perl -w
+#!/usr/bin/perl -w
 
 # $Id: mitmailcreate.pl,v 1.2 2004-07-29 19:11:52 rbasch Exp $
 
