Index: debathena-attach-9.4.30/add.c
===================================================================
--- debathena-attach-9.4.30.orig/add.c	2008-01-01 17:34:40.000000000 -0500
+++ debathena-attach-9.4.30/add.c	2008-01-01 17:39:30.000000000 -0500
@@ -152,7 +152,7 @@
   if (manpath)
     manpath = strdup(manpath);
   else
-    manpath = strdup("");
+    manpath = strdup(":");
 
   infopath = getenv("INFOPATH");
   if (infopath)
