Index: debathena-mitmailrename-9.4.0/mitmailrename.pl
===================================================================
--- debathena-mitmailrename-9.4.0.orig/mitmailrename.pl	2007-01-30 03:00:32.000000000 -0500
+++ debathena-mitmailrename-9.4.0/mitmailrename.pl	2007-01-30 03:00:32.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/athena/bin/perl -w
+#!/usr/bin/perl -w
 
 # $Id: mitmailrename.pl,v 1.2 2004-07-29 19:11:53 rbasch Exp $
 
