Index: debathena-attachandrun-9.4.28/attachandrun.c
===================================================================
--- debathena-attachandrun-9.4.28.orig/attachandrun.c	2007-02-04 01:45:38.000000000 -0500
+++ debathena-attachandrun-9.4.28/attachandrun.c	2007-02-04 01:45:52.000000000 -0500
@@ -123,7 +123,7 @@
       dup2(fds[1], STDOUT_FILENO);
       if (fds[1] != STDOUT_FILENO)
 	close(fds[1]);
-      execl("/bin/athena/attach", "attach", "-p", locker, (char *) NULL);
+      execl("/bin/attach", "attach", "-p", locker, (char *) NULL);
       fprintf(stderr, "%s: execl failed: %s\n", progname, strerror(errno));
       _exit(1);
     }
