--- /etc/pam.d/schroot.orig     2009-04-20 21:37:46.000000000 -0400
+++ /etc/pam.d/schroot        2009-12-21 18:43:51.000000000 -0500
@@ -28,10 +28,7 @@
 # /etc/shadow entries.
 @include common-auth
 @include common-account
-#@include common-session
-
-# Basic pam_unix session module in place of common-session.
-session required pam_unix.so
+@include common-session
 
 # Sets up user limits, please uncomment and read /etc/security/limits.conf
 # to enable this functionality.
