Index: debathena-xdsc-9.4.0/Makefile.in
===================================================================
--- debathena-xdsc-9.4.0.orig/Makefile.in	2007-07-12 17:55:56.000000002 -0400
+++ debathena-xdsc-9.4.0/Makefile.in	2007-07-12 17:56:55.000000002 -0400
@@ -13,10 +13,11 @@
 libdir=@libdir@
 mandir=@mandir@
 datadir=@datadir@
-appdefsdir=${libdir}/X11/app-defaults
+sysconfdir=@sysconfdir@
+appdefsdir=${sysconfdir}/X11/app-defaults
 
 CC=@CC@
-DEFS=@DEFS@ -DEDSC_PATH=\"$(sbindir)/edsc\"
+DEFS=@DEFS@ -DEDSC_PATH=\"${libdir}/debathena-discuss/edsc\"
 CPPFLAGS=@CPPFLAGS@
 CFLAGS=@CFLAGS@
 X_CFLAGS=@X_CFLAGS@
