Index: debathena-nmh-1.0+athena9.4.34/uip/Makefile.in
===================================================================
--- debathena-nmh-1.0+athena9.4.34.orig/uip/Makefile.in	2007-08-17 00:28:21.000000000 -0400
+++ debathena-nmh-1.0+athena9.4.34/uip/Makefile.in	2007-08-17 00:34:42.000000000 -0400
@@ -250,25 +250,10 @@
 # install links
 install-lcmds:
 	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(bindir)
-	$(top_srcdir)/mkinstalldirs $(DESTDIR)/usr/athena/etc
-	rm -f $(DESTDIR)$(bindir)/flists
-	rm -f $(DESTDIR)$(bindir)/folders
-	rm -f $(DESTDIR)$(bindir)/prev
-	rm -f $(DESTDIR)$(bindir)/next
-	$(LN) $(DESTDIR)$(bindir)/flist  $(DESTDIR)$(bindir)/flists
-	$(LN) $(DESTDIR)$(bindir)/folder $(DESTDIR)$(bindir)/folders
-	$(LN) $(DESTDIR)$(bindir)/show   $(DESTDIR)$(bindir)/prev
-	$(LN) $(DESTDIR)$(bindir)/show   $(DESTDIR)$(bindir)/next
-	rm -f $(DESTDIR)/usr/athena/etc/rcvdist
-	rm -f $(DESTDIR)/usr/athena/etc/rcvpack
-	rm -f $(DESTDIR)/usr/athena/etc/rcvstore
-	rm -f $(DESTDIR)/usr/athena/etc/rcvtty
-	rm -f $(DESTDIR)/usr/athena/etc/slocal
-	ln -s ../libexec/rcvdist $(DESTDIR)/usr/athena/etc/rcvdist
-	ln -s ../libexec/rcvpack $(DESTDIR)/usr/athena/etc/rcvpack
-	ln -s ../libexec/rcvstore $(DESTDIR)/usr/athena/etc/rcvstore
-	ln -s ../libexec/rcvtty $(DESTDIR)/usr/athena/etc/rcvtty
-	ln -s ../libexec/slocal $(DESTDIR)/usr/athena/etc/slocal
+	ln -s flist  $(DESTDIR)$(bindir)/flists
+	ln -s folder $(DESTDIR)$(bindir)/folders
+	ln -s show   $(DESTDIR)$(bindir)/prev
+	ln -s show   $(DESTDIR)$(bindir)/next
 
 # install misc support binaries
 install-misc:
