Index: debathena-mailusage-9.4.0/mailusage.pl
===================================================================
--- debathena-mailusage-9.4.0.orig/mailusage.pl	2007-01-30 02:42:28.000000000 -0500
+++ debathena-mailusage-9.4.0/mailusage.pl	2007-01-30 02:42:28.000000000 -0500
@@ -1,4 +1,4 @@
-#!/usr/athena/bin/perl -w
+#!/usr/bin/perl -w
 
 # $Id: mailusage.pl,v 1.4 2004-09-03 20:40:31 rbasch Exp $
 
